tools/socktap/autotalks.cpp

```diff
@@ -0,0 +1,251 @@
+#include "autotalks.hpp"
+
+#include <vanetza/access/g5_link_layer.hpp>
+#include <vanetza/common/serialization_buffer.hpp>
+#include <vanetza/access/ethertype.hpp>
+#include <pthread.h>
+#include <iostream>
+
+#include <atlk/sdk.h>
+#include <atlk/v2x.h>
+#include <atlk/v2x_service.h>
+#include <atlk/ddm_service.h>
+#include <atlk/wdm.h>
+#include <atlk/dsm.h>
+#include <atlk/wdm_service.h>
+#include <atlk/log_service.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <extern/ref_sys.h>
+#include <extern/target_type.h>
+#include <extern/time_sync.h>
+
+#ifdef __cplusplus
+}
+#endif
+
+#include "autotalks_link.hpp"
+
+namespace vanetza
+{
+namespace autotalks
+{
+
+v2x_socket_t* v2x_socket_ptr;
+bool endRxThread = false;
+
+static pthread_t v2x_rx_thread;
+static uint8_t v2x_rx_buffer[2048];
+
+
+int autotalks_device_init(void)
+{
+  atlk_rc_t rc;
+  const char* arg[] = {"./test", "enx0002ccf00006"};
```

Comment:
black magic?

Reply:
I have put the "enx0002ccf00006" network name in define and added a note to the documentation. I think it is hard-coded somewhere from Autotalks